ecr-login/ecr.go

```diff
@@ -26,7 +27,20 @@ import (
 var notImplemented = errors.New("not implemented")
 
 type ECRHelper struct {
-	ClientFactory api.ClientFactory
+	clientFactory api.ClientFactory
+	logger        *logrus.Logger
```

Comment:
How about using FieldLogger?

Reply:
Does this give any flexibility to redirect/discard the output? I'm not very familiar with logrus personally, but it seems to only support log-writing and field-setting methods: https://pkg.go.dev/github.com/sirupsen/logrus#FieldLogger

Reply:
You can either create a logrus.Logger and then set its Out as you did, or use https://github.com/sirupsen/logrus/blob/fdf1618bf7436ec3ee65753a6e2999c335e97221/hooks/test/test.go#L42.

Reply:
Right. In either of those cases, my code has to be aware that logrus is involved, and take an explicit direct dependency on it.

go-containerregistry and many of its users don't depend on logrus directly today, and might only depend on it through ecr-login. Exposing that dependency through ecr-login's API means you can't change it in the future (if you want to) without breaking folks. By only exposing stdlib interfaces in the API, callers can keep a somewhat tidier dependency graph, and you retain some flexibility for the future. AFAIK nobody has expressed a desire to pass a different formatter.

Reply:
As an example, here's how I intend to use this affordance in `ko` to disable ecr-login's log message, without having to expose logrus as an implementation detail to `ko`.

If this PR is merged, I'd also make a similar change to tools like `cosign`, Tekton, Kaniko, etc., for the same reason.